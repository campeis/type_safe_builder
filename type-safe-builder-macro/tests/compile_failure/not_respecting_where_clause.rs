use type_safe_builder_macro::Builder;

trait TraitForField {}
struct FieldStruct {}
impl TraitForField for FieldStruct {}

#[derive(Builder)]
struct GenericStruct<T>
where
    T: TraitForField,
{
    f1: T,
}

struct StructWithoutRequiredTrait {}
fn main() {
    let _builder = GenericStructBuilder::builder().f1(StructWithoutRequiredTrait {});
}
