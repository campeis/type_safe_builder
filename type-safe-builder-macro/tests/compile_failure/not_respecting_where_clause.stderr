error[E0277]: the trait bound `StructWithoutRequiredTrait: TraitForField` is not satisfied
  --> tests/compile_failure/not_respecting_where_clause.rs:17:52
   |
17 |     let _builder = GenericStructBuilder::builder().f1(StructWithoutRequiredTrait {});
   |                                                    ^^ the trait `TraitForField` is not implemented for `StructWithoutRequiredTrait`
   |
   = help: the trait `TraitForField` is implemented for `FieldStruct`
note: required by a bound in `GenericStructBuilderState::<T, Unset>::f1`
  --> tests/compile_failure/not_respecting_where_clause.rs:10:8
   |
10 |     T: TraitForField,
   |        ^^^^^^^^^^^^^ required by this bound in `GenericStructBuilderState::<T, Unset>::f1`
11 | {
12 |     f1: T,
   |     -- required by a bound in this associated function

error[E0277]: the trait bound `StructWithoutRequiredTrait: TraitForField` is not satisfied
  --> tests/compile_failure/not_respecting_where_clause.rs:17:20
   |
17 |     let _builder = GenericStructBuilder::builder().f1(StructWithoutRequiredTrait {});
   |                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `TraitForField` is not implemented for `StructWithoutRequiredTrait`
   |
   = help: the trait `TraitForField` is implemented for `FieldStruct`
note: required by a bound in `GenericStructBuilder::builder`
  --> tests/compile_failure/not_respecting_where_clause.rs:10:8
   |
7  | #[derive(Builder)]
   |          ------- required by a bound in this associated function
...
10 |     T: TraitForField,
   |        ^^^^^^^^^^^^^ required by this bound in `GenericStructBuilder::builder`
